### PR TITLE
 Return an error when the database does not support an auto-generated key

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

When the database does not support auto-generated key recovery, the error message does not contain the relevant info. So, this PR has returned the error in the client initalisation.

Fixes:https://github.com/ballerina-platform/ballerina-standard-library/issues/3900

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/client.bal [hunk 1]
```diff
@@ -31,7 +32,11 @@ public isolated client class Client {
     #                    `connectionPool` provided, the global connection pool (shared by all clients) will be used
     # + return - An `sql:Error` if the client creation fails
     public isolated function init(string url, string? user = (), string? password = (),
-        Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error? {
+            Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error? {
+        if strings:startsWith(url, "jdbc:sqlserver") && isRequestGeneratedKeysSupportInBatchExecute(options) {
```
Comment (niveathika) [inline ballerina/client.bal:36]:
```suggestion
        if strings:startsWith(url, "jdbc:sqlserver") && isRequestGeneratedKeysSupportsBatchExecute(options) {
```

File: ballerina/client.bal [hunk 2]
```diff
@@ -31,7 +32,11 @@ public isolated client class Client {
...
+            return error sql:ApplicationError("MSSQL does not support the retrieval of auto-generated " +
+                                "keys with batch execute function");
```
Comment (niveathika) [inline ballerina/client.bal:38]:
```suggestion
            return error sql:ApplicationError("Unsupported `requestGeneratedKeys` option for MSSQL database, expected `jdbc:EXECUTE`");
```

File: ballerina/client.bal [hunk 3]
```diff
@@ -156,3 +163,8 @@ isolated function nativeBatchExecute(Client sqlClient, string[]|sql:Parameterize
 returns sql:ExecutionResult[]|sql:Error = @java:Method {
     'class: "io.ballerina.stdlib.java.jdbc.nativeimpl.ExecuteProcessor"
 } external;
+
+isolated function isRequestGeneratedKeysSupportInBatchExecute(Options? options) returns boolean {
+    return (options is () || (options is Options &&
+                (options.requestGeneratedKeys == ALL || options.requestGeneratedKeys == BATCH_EXECUTE)));
```
Comment (niveathika) [inline ballerina/client.bal:169]: Wouldnt it be easier to test the negative scenario? `options.requestGeneratedKeys` can only be execute

Comment (kalaiyarasiganeshalingam) [reply ballerina/client.bal]: Updated in fceb82c

File: ballerina/client.bal [hunk 4]
```diff
@@ -31,7 +32,11 @@ public isolated client class Client {
     #                    `connectionPool` provided, the global connection pool (shared by all clients) will be used
     # + return - An `sql:Error` if the client creation fails
     public isolated function init(string url, string? user = (), string? password = (),
-        Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error? {
+            Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error? {
+        if strings:startsWith(url, "jdbc:sqlserver") && isRequestGeneratedKeysSupportsBatchExecute(options) {
+            return error sql:ApplicationError("Unsupported `requestGeneratedKeys` option for MSSQL database, " +
+                        "expected `jdbc:EXECUTE`or `jdbc:NONE`");
+        }
```
Comment (niveathika) [inline ballerina/client.bal:39]:
```suggestion
            return error sql:ApplicationError("Unsupported `requestGeneratedKeys` option for MSSQL database, " +
                        "expected `jdbc:EXECUTE` or `jdbc:NONE`");
        }
```

**Pull request merged**